smithy-typescript-codegen/src/test/java/software/amazon/smithy/typescript/codegen/integration/AddHttpApiKeyAuthPluginTest.java

```diff
@@ -0,0 +1,150 @@
+            not(containsString("= resolveHttpApiKeyAuthConfig(")));
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/ExampleClient.ts").get(),
+            not(containsString("from \"./middleware/HttpApiKeyAuth\"")));
+
+        // make sure that the import and middleware were not used in the operation
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/commands/GetFooCommand.ts").get(),
+            not(containsString("from \"../middleware/HttpApiKeyAuth\"")));
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/commands/GetFooCommand.ts").get(),
+            not(containsString("this.middlewareStack.use(getHttpApiKeyAuthPlugin(configuration")));
+
+        // make sure that the middleware file was not written
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/middleware/HttpApiKeyAuth/index.ts").isPresent(),
+            is(false));
+    }
+
+    @Test
+    public void httpApiKeyAuthClientWithAllOptionalAuthOperations() {
+        MockManifest manifest = new MockManifest();
+        PluginContext context = PluginContext.builder()
+            .pluginClassLoader(getClass().getClassLoader())
+            .model(Model.assembler()
+                .addImport(getClass().getResource("http-api-key-auth-trait-unused.smithy"))
+                .discoverModels()
+                .assemble()
+                .unwrap())
+            .fileManifest(manifest)
+            .settings(Node.objectNodeBuilder()
+                .withMember("service", Node.from("smithy.example#Example"))
+                .withMember("package", Node.from("example"))
+                .withMember("packageVersion", Node.from("1.0.0"))
+                .build())
+            .build();
+
+        new TypeScriptCodegenPlugin().execute(context);
+
+        // make sure that the config and middleware were not added to the client
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/ExampleClient.ts").get(),
+            not(containsString("= resolveHttpApiKeyAuthConfig(")));
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/ExampleClient.ts").get(),
+            not(containsString("from \"./middleware/HttpApiKeyAuth\"")));
+
+        // make sure that the import and middleware were not used in the operation
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/commands/GetFooCommand.ts").get(),
+            not(containsString("from \"../middleware/HttpApiKeyAuth\"")));
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/commands/GetFooCommand.ts").get(),
+            not(containsString("this.middlewareStack.use(getHttpApiKeyAuthPlugin(configuration")));
+
+        // make sure that the middleware file was not written
+        assertThat(manifest.getFileString(CodegenUtils.SOURCE_FOLDER + "/middleware/HttpApiKeyAuth/index.ts").isPresent(),
+            is(false));
+    }
+
+}
```

Comment:
nit: files should end with an empty line.

Looks like this happens in a few of the test case files as well.